src/random.cpp

```diff
@@ -43,15 +43,24 @@ static void RandFailure()
 
 static inline int64_t GetPerformanceCounter()
 {
-    int64_t nCounter = 0;
-#ifdef WIN32
-    QueryPerformanceCounter((LARGE_INTEGER*)&nCounter);
+    // Read the hardware time stamp counter when available.
+    // See https://en.wikipedia.org/wiki/Time_Stamp_Counter for more information.
+#if defined(_MSC_VER)
+    return __rdtsc();
```

Comment:
Looks like it may not be https://docs.microsoft.com/en-us/cpp/intrinsics/rdtsc Maybe move this inside the x86 ifdef checks as well?